Build fails on Ubuntu 20.04

Trying to build on Ubuntu 20.04 (Focal Fossa) ...

```
$ cmake .
-- The C compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'libgps'
--   Found libgps, version 3.20
-- Checking for module 'libnl-genl-3.0'
--   Found libnl-genl-3.0, version 3.4.0
-- Configuring done
-- Generating done
```

... fails with the following error:

```
[ 46%] Building C object wifi/sender/CMakeFiles/sender.dir/main.c.o
/home/janus/github/opendroneid/opendroneid-core-c/wifi/sender/main.c: In function ‘drone_adopt_gps_data’:
/home/janus/github/opendroneid/opendroneid-core-c/wifi/sender/main.c:232:36: error: invalid operands to binary * (have ‘timespec_t’ {aka ‘struct timespec’} and ‘int’)
  232 |  time_in_tenth = gpsdata->fix.time * 10;
      |                  ~~~~~~~~~~~~~~~~~ ^
      |                              |
      |                              timespec_t {aka struct timespec}
```

I assume this is due to the fact that 20.04 has the updated libgps version 3.20. On Ubuntu 18.04, with libgps version 3.17, everything works well.

--- a/wifi/sender/CMakeLists.txt
+++ b/wifi/sender/CMakeLists.txt
@@ -10,6 +10,9 @@ link_libraries(opendroneid m ${GPS_LIBRARIES} ${NL_LIBRARIES} ${GENL_LIBRARIES})
 include_directories(../../libopendroneid ${GPS_INCLUDE_DIRS} ${NL_INCLUDE_DIRS} ${GENL_INCLUDE_DIRS})
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${GPS_CFLAGS_OTHER} ${NL_CFLAGS_OTHER} ${GENL_CFLAGS_OTHER}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -W -Wno-unused-parameter -std=gnu99 -fno-strict-aliasing -MD -MP -D_GNU_SOURCE")
+if (GPS_VERSION VERSION_LESS 3.20)
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DLIBGPS_OLD")
+endif()
 
 add_executable(sender main.c)
 

--- a/wifi/sender/main.c
+++ b/wifi/sender/main.c
@@ -42,6 +42,12 @@ sw@simonwunderlich.de
 
 #include <opendroneid.h>
 
+/* convert a timespec to a double.
+ * if tv_sec > 2, then inevitable loss of precision in tv_nsec
+ * so best to NEVER use TSTONS()
+ * WARNING replacing 1e9 with NS_IN_SEC causes loss of precision */
+#define TSTONS(ts) ((double)((ts)->tv_sec + ((ts)->tv_nsec / 1e9)))
+
 struct global {
 	char server[1024];
 	char port[16];
@@ -198,7 +204,7 @@ static void drone_adopt_gps_data(ODID_UAS_Data *drone,
 	uint64_t time_in_tenth;
 
 	drone->LocationValid = 1;
-	
+
 	/*
 	*	ALL READOUTS FROM GPSD
 	*/
@@ -229,7 +235,11 @@ static void drone_adopt_gps_data(ODID_UAS_Data *drone,
 	drone->Location.SpeedAccuracy = createEnumSpeedAccuracy(gpsdata->fix.eps > gpsdata->fix.epc ? gpsdata->fix.eps : gpsdata->fix.epc);
 
 	/* Time */
+#ifdef LIBGPS_OLD
 	time_in_tenth = gpsdata->fix.time * 10;
+#else
+	time_in_tenth = TSTONS(&gpsdata->fix.time) * 10;
+#endif
 	drone->Location.TimeStamp = (float)((time_in_tenth % 36000) / 10);
 	drone->Location.TSAccuracy = createEnumTimestampAccuracy((float)gpsdata->fix.ept);
 

